eng/common/scripts/job-matrix/job-matrix-functions.ps1

```diff
@@ -196,18 +260,102 @@ function ProcessIncludes([MatrixConfig]$config, [Array]$matrix)
     return $matrix + $inclusionMatrix
 }
 
-function ProcessImport([System.Collections.Specialized.OrderedDictionary]$matrix, [String]$selection)
+function ParseReplacement([String]$replacement) {
+    $parsed = '', '', ''
+    $idx = 0
+    $escaped = $false
+    $operators = '=', '/'
+    $err = "Invalid replacement syntax, expecting <key>=<value>/<replace>"
+
+    foreach ($c in $replacement -split '') {
+        if ($idx -ge $parsed.Length) {
+            throw $err
+        }
+        if (!$escaped -and $c -in $operators) {
+            $idx++
+        } else {
+            $parsed[$idx] += $c
+        }
+        $escaped = $c -eq '\'
+    }
+
+    if ($idx -lt $parsed.Length - 1) {
+        throw $err
+    }
+
+    $replace = $parsed[2]
+    $replace = $replace -replace '\\=', '='
+    $replace = $replace -replace '\\/', '/'
+
+    return @{
+        "key" = $parsed[0]
+        "value" = $parsed[1]
+        "replace" = $replace
+    }
+}
+
+function ProcessReplace
 {
-    if (!$matrix -or !$matrix.Contains($IMPORT_KEYWORD)) {
-        return $matrix, @(), @{}
+    param(
+        [Array]$matrix,
+        [Array]$replacements,
+        [Hashtable]$displayNamesLookup
+    )
+
+    if (!$replacements) {
+        return $matrix
     }
 
-    $importPath = $matrix[$IMPORT_KEYWORD]
-    $matrix.Remove($IMPORT_KEYWORD)
+    $replaceMatrix = @()
+
+    foreach ($element in $matrix) {
+        $replacement = [MatrixParameter[]]@()
+
+        foreach ($perm in $element._permutation) {
+            $replace = $perm
+
+            # Iterate nested permutations or run once for singular values (int, string, bool)
+            foreach ($flattened in $perm.Flatten()) {
+                foreach ($query in $replacements) {
+                    $parsed = ParseReplacement $query
+                    if ($flattened.Name -match $parsed.key -and $flattened.Value -match $parsed.value) {
+                        $perm.Set($parsed.replace, $parsed.key)
```

Comment:
Based on our discussion it would be nice to have this be a replace operator.

Reply:
Added. Take a look at the readme/test samples.